### PR TITLE
Improve aggregator documentation

### PR DESCRIPTION
**Pull request by ed-xmos**

### CODE REVIEW

File: doc/programming_guide/mic_aggregator/mic_aggregator.rst [hunk 1]
```diff
@@ -1,3 +1,248 @@
 .. _sln_voice_mic_aggregator_programming_guide:
 
 .. include:: ../../../examples/mic_aggregator/README.rst
+
+
+Operation
+=========
+
+The design consists of a number of tasks connected via the xcore-ai silicon communication channels.
+The decimators in the microphone array are configured to produce a 48 kHz PCM output.
```
Comment (xhuw) [inline doc/programming_guide/mic_aggregator/mic_aggregator.rst:10]: I can see that this application relies on `MyTwoStageDecimator` from the (as far as I can see) undocumented `par_decimator` demo in the lib_mic_array repo. 

Having one demo rely on another feels like spaghetti in the making. It would be better to upgrade `MyTwoStageDecimator` to be a proper part of lib_mic_array that has a useful name and is part of the `lib_mic_array` cmake target. Then modify both demos to use it properly.

I can see that isn't really the scope of this review though so just highlighting what I noticed.

File: doc/programming_guide/mic_aggregator/mic_aggregator.rst [hunk 2]
```diff
@@ -1,3 +1,248 @@
...
+
+Board Configuration
+===================
+
+Make the following connections using flying leads:
+
+===============     ================    ==============================================
+Host Connection     Board Connection    Note
+===============     ================    ==============================================
+MIC CLK             J14 ‘00’            This is the microphone clock which is to be sent to the PDM microphones from J14.
```
Comment (xhuw) [inline doc/programming_guide/mic_aggregator/mic_aggregator.rst:152]: It took me a while to decipher what "J14 '00'" means. Maybe it would make more sense if I have the board in front of me? "J14 pin 0" could be clearer

**Review by xhuw — APPROVED** [review]

File: doc/programming_guide/mic_aggregator/mic_aggregator.rst (hunk 1 above)
Comment (ed-xmos) [reply doc/programming_guide/mic_aggregator/mic_aggregator.rst:10]: Sorry I missed this before - this is a very good point. I followed the design idiom that mic_array set but agree this would be a helpful update. Mic array in general is quite hard to follow (in my opinion) and anything that tidies that up is useful. I'll create an issue.

Comment (ed-xmos) [reply doc/programming_guide/mic_aggregator/mic_aggregator.rst:10]: https://github.com/xmos/lib_mic_array/issues/210
